Test/Test/Source.cpp

```diff
@@ -1,78 +1,72 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+//#include <stdio.h>
+
+
 #include "ConsoleReader.h"
+#include "CharOccurrences.h"
 
-const int possibleChars = 128;
+#include <log4cpp/Portability.hh>
+#include <log4cpp/PropertyConfigurator.hh>
+#include <log4cpp/Category.hh>
+#include <log4cpp/Appender.hh>
+#include <log4cpp/FileAppender.hh>
+#include <log4cpp/OstreamAppender.hh>
+#include <log4cpp/Layout.hh>
+#include <log4cpp/BasicLayout.hh>
+#include <log4cpp/Priority.hh>
+#include <log4cpp/NDC.hh>
 
-void calculateOccurs(int arr[], std::string line) {
-	for (int j = 0; j < line.length(); ++j) {
-		++arr[int(line[j])];
-	}
-}
 
 
-bool calculateOccursInFile(std::string path, int arr[]) {
-	std::string line;
-	std::ifstream input;
-	input.open(path);
-	if (!input.is_open()) {
-		std::cout << "Could not open the file with a path " << path;
-		return false;
-	}
-	while (!input.eof()) {
-		getline(input, line);
-		// std::cout << line << std::endl;
 
-		// calculates the number of occurrences of each unique symbol
-		calculateOccurs(arr, line);
-	}
-	input.close();
-	return true;
-}
-// so this is not really reusable, and I'm not sure if that can be fixed, or if it even is necessary
+int main() {
+	// Logging stuff
+	std::string initFileName = "log4cpp.properties";
+	log4cpp::PropertyConfigurator::configure(initFileName);
 
+	log4cpp::Category& root = log4cpp::Category::getRoot();
 
-void outputToFile(std::string outputPath, int arr[]) {
-	std::ofstream output;
-	output.open(outputPath);
+	log4cpp::Category& main =
+		log4cpp::Category::getInstance(std::string("main"));
 
-	std::cout << "Creating output.txt file with char occurences in input file\n";
-	for (int i = 0; i < possibleChars; ++i) {
-		if (arr[i]) {
-			// std::cout << char(i) << ": " << charOccurs[i] << std::endl;
-			output << char(i) << ": " << arr[i] << std::endl;
-		}
-	}
-	output.close();
-}
+	main.info("Started app");
 
 
-int main() {
 	// gets the path to file as an input from command line
-	std::string path;
+	std::string inputPath;
 	ConsoleReader reader;
-	std::cout << "Input file path:\n";
-	if (reader.readConsole()) {
-		path = reader.buffer;
+	std::string message = "Input file path:";
+	if (reader.readConsole(message)) {
```

Comment:
BTW, can we have char* parameter instead of string? In this case we can directly call like reader.readConsole("jdkfjkwefkj")

Reply:
Is this a tricky question or can I actually do that :P ?

Reply:
Huh, I guess we can